src/Mvc/Mvc.Core/src/ApplicationModels/InferParameterBindingInfoConvention.cs

```diff
@@ -95,6 +113,11 @@ internal BindingSource InferBindingSourceForParameter(ParameterModel parameter)
     {
         if (IsComplexTypeParameter(parameter))
         {
+            if (_serviceProviderIsService?.IsService(parameter.ParameterType) == true)
```

Comment:
Super nit because you get 💯 points

```suggestion
            if (_serviceProviderIsService?.IsService(parameter.ParameterType) is true)
```